lib/services/firebase_service.dart

```diff
@@ -742,7 +742,77 @@ class FirebaseService extends IFirebaseService {
       print('$onError Cannot find user');
     });
     var anSubCollection = await getLatestAnSubCollection(docId: userId);
-    var patientState = anSubCollection['state'];
     return anSubCollection['state'];
   }
+
+  Future<Map<String, dynamic>> getPatientDetail({@required String hn}) async {
+    var usersCollection = await _firestore
+        .collection('Users')
+        .where('hn', isEqualTo: hn)
+        .get()
+        .then((value) {
+      var map = value.docs.first.data();
+      map['id'] = value.docs.first.id;
+      return map;
+    });
+    var anlength = usersCollection['an'].length - 1;
+    String an = usersCollection['an'][anlength]['an'];
+    var anSubCollection = await _firestore
+        .collection('Users')
+        .doc(usersCollection['id'])
+        .collection('an')
+        .doc(an)
+        .get()
+        .then((value) => value.data())
+        .catchError((onError) {
+      print('$onError');
+    });
+    var generalForm = await getFormListInAnBasedOnState(
+        userId: usersCollection['id'],
+        patientState: anSubCollection['state'],
+        formName: 'General');
+    var generalFormData = Map();
+    if (generalForm.isNotEmpty)
+      generalFormData = await _firestore
+          .collection('Forms')
+          .doc(generalForm.last['formId'])
+          .get()
+          .then((value) => value.data())
+          .catchError((onError) {
+        print('$onError Cann\'t find General Form on ${usersCollection['hn']}');
+      });
+
+    DateFormat dateFormatter = DateFormat('dd-MM-yyyy');
+    var map = {
+      'fullName': '${usersCollection['name']} ${usersCollection['surname']}',
+      'gender': usersCollection['gender'] ?? '-',
+      'age': _calculationService
+          .calculateAge(birthDate: usersCollection['dob'].toDate())
+          .toString(),
+      'dob': dateFormatter.format(usersCollection['dob'].toDate()) ?? '-',
+      'patientTel': usersCollection['patientTel'] ?? '-',
+      'address': usersCollection['address'] ?? '-',
+      'height': anSubCollection['height'].toString() ?? '-',
+      'weight': anSubCollection['oldWeight'].toString() ?? '-',
+      'bwl': _calculationService
+          .calculateBML(
+              oldWeight: anSubCollection['oldWeight'],
```

Comment:
เหมือนว่าเราจะต้องเปลี่ยนมาดึงold weight ผ่าน previsit form 
<img width="374" alt="Screen Shot 2564-04-24 at 18 42 31" src="https://user-images.githubusercontent.com/57941667/115957458-d673f600-a52c-11eb-8e78-7e5f9803548f.png">
Because previously we mock old weight data in an collection

Reply:
@PormpimonPang  Gonna change the plan. Things inserted through `Pre-Visit` gonna be added to anSubCollection as well.
Searching for forms uses a lot of resourse.